src/EnergyPlus/HVACControllers.cc

```diff
@@ -2324,52 +2241,33 @@ namespace HVACControllers {
     //        End of Update subroutines for the Controller Module
     // *****************************************************************************
 
-    // Beginning of Reporting subroutines for the Controller Module
-    // *****************************************************************************
-
-    void ReportController(int const EP_UNUSED(ControlNum)) // unused1208
+    void CheckTempAndHumRatCtrl(int const ControlNum, bool &IsConvergedFlag)
     {
 
-        // SUBROUTINE INFORMATION:
-        //       AUTHOR         <author>
-        //       DATE WRITTEN   <date_written>
-        //       MODIFIED       na
-        //       RE-ENGINEERED  na
-
-        // PURPOSE OF THIS SUBROUTINE:
-        // This subroutine needs a description.
-
-        // METHODOLOGY EMPLOYED:
-        // Needs description, as appropriate.
-
-        // REFERENCES:
-        // na
-
-        // USE STATEMENTS:
-        // na
-
-        // Locals
-        // SUBROUTINE ARGUMENT DEFINITIONS:
-
-        // SUBROUTINE PARAMETER DEFINITIONS:
-        // na
-
-        // INTERFACE BLOCK SPECIFICATIONS
-        // na
-
-        // DERIVED TYPE DEFINITIONS
-        // na
-
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        // na
-
-        // Still needs to report the Controller power from this component
-        // Write(*,*)=ControllerProps(ControlNum)%ControllerPower
+        {
+            auto &thisController(ControllerProps(ControlNum));
+            if (IsConvergedFlag) {
+                if (thisController.ControlVar == iTemperatureAndHumidityRatio) {
+                    // For temperature and humidity control, after temperature control is converged, check if humidity setpoint is met
+                    if (!thisController.HumRatCtrlOverride) {
+                        // For humidity control tolerance, always use 0.0001 which is roughly equivalent to a 0.015C change in dewpoint
+                        if (Node(thisController.SensedNode).HumRat > (Node(thisController.SensedNode).HumRatMax + 1.0e-4)) {
+                            // Turn on humdity control and restart controller
+                            IsConvergedFlag = false;
+                            thisController.HumRatCtrlOverride = true;
+                            if (thisController.Action == iReverseAction) {
```

Comment:
This seems redundant but I'm OK with it. Controller should spit out warnings that it can't control temperature much less humidity.